ml-proto/test/float32.wasm

```diff
@@ -0,0 +1,52 @@
+(module
+  (func $eq_float32 (param $x f32) (param $y f32) (result i32)
+    (eq.f32 (getlocal $x) (getlocal $y))
+  )
+
+  (func $eq_float64 (param $x f64) (param $y f64) (result i32)
+    (eq.f64 (getlocal $x) (getlocal $y))
+  )
+
+  (func $div_float32 (param $x f32) (param $y f32) (result f32)
+    (div.f32 (getlocal $x) (getlocal $y))
+  )
+
+  (func $div_float64 (param $x f64) (param $y f64) (result f64)
+    (div.f64 (getlocal $x) (getlocal $y))
+  )
+
+  (export "eq_float32" $eq_float32)
+  (export "eq_float64" $eq_float64)
+  (export "div_float32" $div_float32)
+  (export "div_float64" $div_float64)
+)
+
+(asserteq
+  (invoke "eq_float32" (const.f32 1.123456789012345) (const.f32 1.123456789))
```

Comment:
Hex float is perfectly readable / writable by humans :-)
More seriously, I think it's pretty important that text parsers reject numbers which aren't precisely representable. It also important that all parsers know exactly how to do this so that they all accept the same inputs! Hex float is the easiest way to get this, but I'm open to supporting more.

I'd like to make sure that we discuss any other representation in https://github.com/WebAssembly/design/issues/292, and that we write down the outcome so it's easy to search in the future. These PR discussions aren't searchable!